rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy

```diff
@@ -1194,8 +1221,38 @@ class ScheduledExecutionService implements ApplicationContextAware{
         //TODO:validate inputs
         return [failed:false]
     }
-    
-    
+
+    def _doUpdateExecutionFlags(params, user, String roleList, Framework framework, AuthContext authContext, changeinfo = [:]) {
+        log.debug("ScheduledExecutionController: update : attempting to updateExecutionFlags: " + params.id + ". params: " + params)
+
+        def ScheduledExecution scheduledExecution = getByIDorUUID(params.id)
+        if (!scheduledExecution) {
+            return [success: false]
+        }
+
+        if (!frameworkService.authorizeProjectJobAll(authContext, scheduledExecution, [AuthConstants.ACTION_UPDATE], scheduledExecution.project)) {
+            return [success: false, scheduledExecution: scheduledExecution, message: "Update Job ${scheduledExecution.extid}", unauthorized: true]
+        }
+
+        def oldSched = scheduledExecution.scheduled
+        def oldJobName = scheduledExecution.generateJobScheduledName()
+        def oldJobGroup = scheduledExecution.generateJobGroupName()
+
+        if (null != params.scheduleEnabled) {
+            scheduledExecution.properties.scheduleEnabled = params.scheduleEnabled
+        }
+
+        if (null != params.executionEnabled) {
+            scheduledExecution.properties.executionEnabled = params.executionEnabled
+        }
+
+        if (!scheduledExecution.validate()) {
+            return [success: false]
+        }
```

Comment:
should call `scheduledExecution.save()` explicitly, so that any problems saving it are reflected in response.  otherwise it will cause an error while flushed with the hibernate session, and that won't be reflected in the response